src/highdicom/seg/sop.py

```diff
@@ -443,20 +446,21 @@ def __init__(
                     )
                 )
 
-        if is_multiframe:
-            if self._coordinate_system == CoordinateSystemNames.SLIDE:
-                source_plane_orientation = PlaneOrientationSequence(
-                    coordinate_system=self._coordinate_system,
-                    image_orientation=src_img.ImageOrientationSlide
-                )
-            else:
-                src_sfg = src_img.SharedFunctionalGroupsSequence[0]
-                source_plane_orientation = src_sfg.PlaneOrientationSequence
-        else:
+        if is_relative_to_slide:
             source_plane_orientation = PlaneOrientationSequence(
                 coordinate_system=self._coordinate_system,
-                image_orientation=src_img.ImageOrientationPatient
+                image_orientation=src_img.ImageOrientationSlide
             )
+        else:
+            if is_multiframe:
+                src_sfg = src_img.SharedFunctionalGroupsSequence[0]
+                source_plane_orientation = src_sfg.PlaneOrientationSequence
+            else:
+                source_plane_orientation = PlaneOrientationSequence(
+                    coordinate_system=self._coordinate_system,
+                    image_orientation=src_img.ImageOrientationPatient
+                )
```

Comment:
Could this branching be simplified a bit by having a variable holding `image_orientation` something like this?

```python
if is_multiframe:
    src_sfg = src_img.SharedFunctionalGroupsSequence[0]
    source_plane_orientation = src_sfg.PlaneOrientationSequence
else:
    image_orientation = src_img.ImageOrientationSlide if is_relative_to_slide else src_img.ImageOrientationPatient
    source_plane_orientation = PlaneOrientationSequence(
        coordinate_system=self._coordinate_system,
        image_orientation=image_orientation
    )
```

Reply:
I wondered this, but if I'm not mistaken the current implementation doesn't account for that either?

Reply:
My understanding is as follows:

In case of a slide coordinate system, the information is always stored in `ImageOrientationSlide` irrespective of whether the source image is multi-frame or single-frame. However, that is one of the aspects I am currently discussing with @dclunie as part of the mentioned correction proposal.

In case of a patient coordinate system, the information is stored either in `PlaneOrientationSequence` if the source image is multi-frame or in `ImageOrientationPatient` if the source image is single-frame.

Does the current implementation not match this?

Reply:
Ah I see, makes sense